### ISSUE TITLE
[Bug report] - Passive extraction tries saving None data after error

### ISSUE THREAD

**Issue by k1o0** [int-brain-lab/iblenv#232]
**Describe the bug**
From a [passive extraction log](https://alyx.internationalbrainlab.org/admin/jobs/task/3dcdb3a3-18bb-4f2b-b145-4fded86fe901/change/):
```
2021-04-19 10:30:47,231,231 ERROR [ephys_passive.py:528] Failed to extract passive periods: The number of expected spacer (3) is different than the one found on the raw trace (4.0)
2021-04-19 10:30:47,231,231 ERROR [tasks.py:85] Traceback (most recent call last):
File "/home/ibladmin/Documents/PYTHON/envs/iblenv/lib/python3.7/site-packages/ibllib/pipes/tasks.py", line 82, in run
self.outputs = self._run(**kwargs)
File "/home/ibladmin/Documents/PYTHON/envs/iblenv/lib/python3.7/site-packages/ibllib/pipes/ephys_preprocessing.py", line 423, in _run
data, paths = ephys_passive.PassiveChoiceWorld(self.session_path).extract(save=True)
File "/home/ibladmin/Documents/PYTHON/envs/iblenv/lib/python3.7/site-packages/ibllib/io/extractors/base.py", line 48, in extract
files = self._save(out, path_out=path_out) if save else None
File "/home/ibladmin/Documents/PYTHON/envs/iblenv/lib/python3.7/site-packages/ibllib/io/extractors/base.py", line 90, in _save
_write_to_disk(fpath, data)
File "/home/ibladmin/Documents/PYTHON/envs/iblenv/lib/python3.7/site-packages/ibllib/io/extractors/base.py", line 74, in _write_to_disk
data.to_csv(file_path, sep=sep)
AttributeError: 'NoneType' object has no attribute 'to_csv'
```

The first error is caught by the passive extraction _extract method which them returns Nones.  The AttributeError arises from trying to save None data.  

There is already a try/except in the calling Task class and the passive ephys is already in its own task so there's no reason to have an extra try except.

**Comment by k1o0** [comment]
Possible solutions:

1. Remove the try/except.  I see they currently allow registration of partial datasets.  Are partial passive datasets useful?  If not remove them.
2. Set the missing data to None in the `save_names` attribute.  These will then be skipped.
3. Deal with it further downstream, in _save

**Comment by nbonacchi** [comment]
Should be fixed now with PR#316, closing for now
